concordium-node/src/bin/cli.rs

```diff
@@ -138,10 +140,49 @@ async fn main() -> anyhow::Result<()> {
         None
     };
 
-    if let Some(ref import_path) = conf.cli.baker.import_path {
-        info!("Starting out of band catch-up");
-        consensus.import_blocks(import_path.as_bytes());
-        info!("Completed out of band catch-up");
+    // Out-of-band catch-up
+    if let Some(ref import_blocks_from) = conf.cli.baker.import_blocks_from {
+        let from: anyhow::Result<String> = if let Ok(import_url) = Url::parse(import_blocks_from) {
+            let default_filename = format!(
+                "{}_{}{}",
+                config::CATCHUP_FILE_BASENAME,
+                Utc::now().timestamp(),
+                config::CATCHUP_FILE_EXT
+            );
+            let filename = import_url
+                .path_segments()
+                .and_then(|x| x.last())
+                .map(|x| x.to_string())
+                .unwrap_or(default_filename);
+            let import_path = data_dir_path.to_path_buf().join(&filename);
+
+            info!(
+                "Downloading the catch-up file from {} to {}",
+                import_url,
+                import_path.display().to_string()
+            );
+            let file = std::fs::File::create(&import_path)?;
+            let mut buffer = std::io::BufWriter::new(file);
+            let mut stream = reqwest::get(import_url).await?.bytes_stream();
+            while let Some(Ok(bytes)) = stream.next().await {
+                buffer.write_all(&bytes)?;
+            }
+            buffer.flush()?;
```

Comment:
The node will now stop if any of these result in an error, since these are no longer in a closure.
We don't want the node to stop, but just log some error.